qube/quantitative/ta/indicators.py

```diff
@@ -1447,3 +1448,27 @@ def psar(ohlc, iaf=0.02, maxaf=0.2):
     psar_i, psarbear, psarbull = __psar(ohlc['close'].values, ohlc['high'].values, ohlc['low'].values, iaf, maxaf)
 
     return pd.DataFrame({"psar": psar_i, "up": psarbear, "down": psarbull}, index=ohlc.index)
+
+def fdi(x, e_period = 30):
+    fdi_result = pd.DataFrame()
+    
+    lastBars = x.count().min() - e_period
```

Comment:
Don't we need to check if x is instance of Series before calling methods like .count() etc ?
What if x is np.array of just a list of values ?

Reply:
refactoring at all